eth/v1alpha1/block.proto

```diff
@@ -0,0 +1,203 @@
+}
+
+// Deposit into the Ethereum 2.0 from the Ethereum 1.x deposit contract.
+message Deposit {
+    message Data {
+        // 48 byte BLS public key of the validator.
+        bytes public_key = 1;
+
+        // A 32 byte hash of the withdrawal address public key.
+        bytes withdrawal_credentials = 2;
+
+        // Deposit amount in gwei.
+        uint64 amount = 3;
+
+        // 96 byte signature from the validators public key.
+        bytes signature = 4;
+    }
+
+    // 32 byte roots in the deposit tree branch.
+    repeated bytes proof = 1;
+
+    Data data = 2;
+}
+
+// A message that represents a validator signaling that they want to voluntarily
+// withdraw from the active validator set.
+message VoluntaryExit {
+    // The epoch on when exit request becomes valid.
+    uint64 epoch = 1; 
+
+    // Index of the exiting validator.
+    uint64 validator_index = 2;
+
+    // Validator's 96 byte signature
+    bytes signature = 3;
+}
+
+// A beacon chain transfer is a ETH currency transfer between two validators.
+message Transfer {
+    // Validator index of the sender.
+    uint64 sender_index = 1;
+
+    // Validator index of the recipient. 
+    uint64 recipient_index = 2;
+
+    // Amount in gwei sent to the recipient.
+    uint64 amount = 3; 
+
+    // Fee in gwei for the block proposer to include this transfer.
+    uint64 fee = 4;
+
+    // Slot which this transfer was included in a block.
+    uint64 inclusion_slot = 5;
+
+    // 48 byte sender's withdrawal public key.
+    bytes sender_withdrawal_public_key = 6;
+
+    // 96 byte signature from the sender's withdrawal key.
+    bytes signature = 7;
+}
+
+// Eth1Data represents references to the Ethereum 1.x deposit contract.
+message Eth1Data {
+    // The 32 byte block hash of the Ethereum 1.x block considered for deposit 
+    // inclusion.
+    bytes block_hash = 1;
+
+    // The 32 byte deposit tree root for the last deposit included in this
+    // block.
+    bytes deposit_root = 2;
+
+    // The total number of deposits included in the beacon chain since genesis
+    // including the deposits in this block.
+    uint64 deposit_count = 3;
+}
+
+// A beacon block header is essentially a beacon block with only a reference to
+// the beacon body as a 32 byte merkle tree root. This type of message is more
+// lightweight than a full beacon block.
+message BeaconBlockHeader {
+    // Beacon chain slot that this block represents.
+    uint64 slot = 1; 
+
+    // 32 byte merkle tree root of the parent ssz encoded block.
+    bytes parent_root = 2;
+
+    // 32 byte merkle tree root of the resulting ssz encoded state after processing this block.
+    bytes state_root = 3;
+
+    // 32 byte merkle tree root of the ssz encoded block body.
+    bytes body_root = 4;
+
+    // 96 byte BLS signature from the validator that produced this block.
+    bytes signature = 5;
+}
+
+message IndexedAttestation {
+    // TODO(preston): what are these?
```

Comment:
I believe it's for the phase 1 custody game between validators on shard chains. According to the validator spec [it is a stub for phase 0](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/validator/0_beacon-chain-validator.md#custody-bitfield)